.golangci.yml

```diff
@@ -1,3 +1,27 @@
 run:
   deadline: 5m
   allow-parallel-runners: true
+
+linters:
+  enable:
+    - dupl
+    - errcheck
+    - exportloopref
+    - goconst
+    - gocyclo
+    - gofmt
+    - goimports
+    - gosimple
+    - govet
+    - ineffassign
+    - misspell
+    - staticcheck
+    - typecheck
+    - unconvert
+    - unparam
+    - unused
+    #FIXME: Required fix the errors to allow enable the following checks
+    #- lll
+    #- nakedret
+    #- revive
```

Comment:
Other than the `goconst, nakedret, revive` this list looks reasonable.

In more detail:
- the `lll` needs to be used with `MAXLENGTH=120`; IIRC that's our unilaterally agreed length limit (but may be 150).
- the `goconst, nakedret` simply go against our practices.
- IIUC the `revive` is a separate linter which kind of duplicates the `golint`; not sure if the benefit is integrating it is worth the effort.

Reply:
> the lll needs to be used with MAXLENGTH=120; IIRC that's our unilaterally agreed length limit (but may be 150).
> the goconst, nakedret simply go against our practices.

Totally fine 👍 
I will change that. 

> IUC the revive is a separate linter which kind of duplicates the golint; not sure if the benefit is integrating it is worth the effort.

It will check things like - var-naming conventions
- unused-parameter and code 
- this one will need to be fixed a lot so I am okay with not adding either 

See the current errors: 

```shell
 var-naming: don't use ALL_CAPS in Go names; use CamelCase (revive)
const GIT_CREDS_HELPER = "git-credential-fio"
      ^
subcommands/git/cmd.go:66:17: unused-parameter: parameter 'cmd' seems to be unused, consider removing or renaming it as _ (revive)
func doGitCreds(cmd *cobra.Command, args []string) {
                ^
subcommands/targets/show.go:29:7: var-naming: don't use underscores in Go names; const application_spdx should be applicationSpdx (revive)
const application_spdx = "application/application_spdx.json"
      ^
subcommands/targets/show.go:116:2: var-naming: var shownCiUrl should be shownCiURL (revive)
	shownCiUrl := false
	^
subcommands/targets/deltas.go:100:36: error-strings: error strings should not be capitalized or end with punctuation or a newline (revive)
		subcommands.DieNotNil(errors.New("No targets found to generate deltas for."))
		                                 ^
subcommands/targets/edit.go:100:2: var-naming: var jobservUrl should be jobservURL (revive)
	jobservUrl, webUrl, err := api.TargetsPut(factory, content)
	^
subcommands/targets/add.go:114:71: var-naming: func parameter hwIdToHashPairs should be hwIDToHashPairs (revive)
func createOstreeTarget(factory string, tags []string, srcTag string, hwIdToHashPairs []string) (Targets, error) {
                                                                      ^
subcommands/targets/artifacts.go:112:18: unused-parameter: parameter 'cmd' seems to be unused, consider removing or renaming it as _ (revive)
func doArtifacts(cmd *cobra.Command, args []string) {
                 ^
subcommands/targets/add.go:121:2: var-naming: var hwIdToHash should be hwIDToHash (revive)
	hwIdToHash := map[string]interface{}{}
	^
subcommands/targets/add.go:122:6: var-naming: var curHwId should be curHwID (revive)
	var curHwId string
	    ^
subcommands/targets/add.go:148:2: var-naming: var latestTargetsPerHwId should be latestTargetsPerHwID (revive)
	latestTargetsPerHwId := make(map[string]*client.Target) // latest Targets per hardware ID that have  `srcTag`
	^
subcommands/targets/add.go:158:3: var-naming: var curTargetHwId should be curTargetHwID (revive)
		curTargetHwId := target.HardwareId()
		^
subcommands/targets/offline-update.go:188:3: increment-decrement: should replace ver += 1 with ver++ (revive)
		ver += 1
		^
subcommands/targets/tail.go:24:13: unused-parameter: parameter 'cmd' seems to be unused, consider removing or renaming it as _ (revive)
func doTail(cmd *cobra.Command, args []string) {
            ^
subcommands/targets/tag.go:116:2: var-naming: var jobServUrl should be jobServURL (revive)
	jobServUrl, webUrl, err := api.TargetUpdateTags(factory, updates)
	^
subcommands/targets/tests.go:87:39: var-naming: func parameter testId should be testID (revive)
func show(factory string, target int, testId string) {
                                      ^
subcommands/targets/tests.go:136:3: var-naming: var testId should be testID (revive)
		testId := args[1]
		^
subcommands/targets/tests.go:140:3: var-naming: var testId should be testID (revive)
		testId := args[1]
		^
subcommands/targets/image.go:56:2: var-naming: var jobServUrl should be jobServURL (revive)
	jobServUrl, webUrl, err := api.TargetImageCreate(factory, inputTarget, appsShortlist, ciScriptsRepo, ciScriptsRef)
	^
subcommands/targets/prune.go:81:6: var-naming: don't use underscores in Go names; var target_names should be targetNames (revive)
	var target_names []string
	    ^
subcommands/targets/deltas.go:18:2: var-naming: var hwId should be hwID (revive)
	hwId   string
	^
subcommands/targets/deltas.go:115:2: var-naming: var jobServUrl should be jobServURL (revive)
	jobServUrl, webUrl, err := api.TargetDeltasCreate(factory, toVer, froms, hwId)
	^
subcommands/targets/prune.go:148:2: var-naming: var jobservUrl should be jobservURL (revive)
	jobservUrl, webUrl, err := api.TargetDeleteTargets(factory, target_names)
	^
subcommands/login/cmd.go:54:3: var-naming: var apiUrl should be apiURL (revive)
		apiUrl := "https://" + strings.Replace(u.Host, "app", "api", 1)
		^
subcommands/login/cmd.go:61:3: var-naming: var credsUrl should be credsURL (revive)
		credsUrl := fmt.Sprintf("https://%s/settings/credentials/", u.Host)
		^
subcommands/login/cmd.go:92:21: var-naming: func parameter credsUrl should be credsURL (revive)
func promptForCreds(credsUrl string) (string, string) {
                    ^
subcommands/login/cmd.go:102:2: var-naming: var clientId should be clientID (revive)
	clientId := strings.Trim(scanner.Text(), " ")
	^
subcommands/teams/cmd.go:47:42: var-naming: don't use underscores in Go names; func parameter team_name should be teamName (revive)
func doGetTeam(api *client.Api, factory, team_name string) {
                                         ^
subcommands/waves/rollout.go:203:4: increment-decrement: should replace i += 1 with i++ (revive)
			i += 1
			^
subcommands/waves/init.go:71:2: var-naming: don't use underscores in Go names; var new_targets should be newTargets (revive)
	new_targets, err := api.TargetsList(factory, version)
	^
subcommands/waves/init.go:77:2: var-naming: don't use underscores in Go names; var current_targets should be currentTargets (revive)
	current_targets, err := api.ProdTargetsGet(factory, tag, false)
	^
subcommands/waves/list.go:23:38: unused-parameter: parameter 'args' seems to be unused, consider removing or renaming it as _ (revive)
func doListWaves(cmd *cobra.Command, args []string) {
                                     ^
subcommands/waves/init.go:190:36: error-strings: error strings should not be capitalized or end with punctuation or a newline (revive)
		subcommands.DieNotNil(fmt.Errorf(`Root role is not configured to sign targets offline.
Please, run "fioctl keys tuf rotate-offline-key --role=targets" in order to create offline targets keys.`))
subcommands/waves/rollout.go:193:14: var-declaration: should drop = 0 from declaration of var i; it is the zero value (revive)
	var i int = 0
	            ^
subcommands/waves/init.go:238:9: indent-error-flow: if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary) (revive)
	} else {
		return target.Custom.UnmarshalJSON(data)
	}
subcommands/common_config.go:23:2: var-naming: don't use ALL_CAPS in Go names; use CamelCase (revive)
	FIO_TOML_NAME        = "z-50-fioctl.toml"
	^
subcommands/common_config.go:24:2: var-naming: don't use ALL_CAPS in Go names; use CamelCase (revive)
	FIO_COMPOSE_APPS_DIR = "/var/sota/compose-apps"
	^
subcommands/common_config.go:81:32: var-naming: method GetUsersLookupById should be GetUsersLookupByID (revive)
func (opts *LogConfigsOptions) GetUsersLookupById(api *client.Api, factory string) {
                               ^
subcommands/common_config.go:113:7: increment-decrement: should replace listLimit -= 1 with listLimit-- (revive)
			if listLimit -= 1; listLimit == 0 {
			   ^
subcommands/common_config.go:115:11: indent-error-flow: if block ends with a return statement, so drop this else and outdent its block (revive)
			} else {
				fmt.Println("")
			}
subcommands/devices/updates_list.go:45:4: increment-decrement: should replace listLimit -= 1 with listLimit-- (revive)
			listLimit -= 1
			^
subcommands/devices/delete.go:30:10: superfluous-else: if block ends with call to os.Exit function, so drop this else and outdent its block (move short variable declaration to its own line if necessary) (revive)
		} else {
			fmt.Printf("ok\n")
		}
subcommands/devices/config_set.go:119:11: indent-error-flow: if block ends with a return statement, so drop this else and outdent its block (revive)
			} else {
				return api.DevicePatchConfig(factory, device.Name, cfg, false)
			}
subcommands/devices/list.go:25:2: var-naming: var deviceUuid should be deviceUUID (revive)
	deviceUuid          string
	^
subcommands/devices/list.go:220:2: var-naming: don't use underscores in Go names; var name_ilike should be nameIlike (revive)
	name_ilike := ""
	^
subcommands/devices/config_wireguard.go:112:2: var-naming: var serverIp should be serverIP (revive)
	serverIp, err := ipToUint32(wsc.VpnAddress)
	^
subcommands/keys/tuf_rotate_all_keys.go:45:42: unused-parameter: parameter 'unusedArgs' seems to be unused, consider removing or renaming it as _ (revive)
func doRotateAllKeys(cmd *cobra.Command, unusedArgs []string) {
                                         ^
subcommands/keys/tuf_updates_delete_offline_key.go:53:55: unused-parameter: parameter 'args' seems to be unused, consider removing or renaming it as _ (revive)
func doTufUpdatesDeleteOfflineKey(cmd *cobra.Command, args []string) {
                                                      ^
subcommands/keys/tuf_updates_review.go:31:45: unused-parameter: parameter 'args' seems to be unused, consider removing or renaming it as _ (revive)
func doTufUpdatesReview(cmd *cobra.Command, args []string) {
                                            ^
subcommands/keys/tuf_copy_targets.go:15:2: redefines-builtin-id: redefinition of the built-in function copy (revive)
	copy := &cobra.Command{
		Use:    "copy-targets <offline key archive> <new targets archive>",
		Short:  "Copy the target signing credentials from the offline key archive",
		Hidden: true,
		Run:    doCopyTargets,
		Args:   cobra.ExactArgs(2),
		Long: `This command extracts the target signing credentials required for initializing 
waves into a new tarball so that the offline key archive isn't required for
rolling out production updates. This should be run after each target key
rotation and distributed to the operator in charge of production OTAs.`,
		Deprecated: `it will be removed in the future.
Please, use a new approach to rotate the targets key into a separate file:
	fioctl keys tuf rotate-offline-key --role=targets \
		--keys=<offline-creds.tgz> --targets-keys=<offline-targets-creds.tgz>
`,
	}
make: *** [lint] Error 1
```

Reply:
IMHO the revive thing is definitely a no-go.
It creates too opinionated warnings, some of which go against a normal coding style.